backend/translation/views.py

```diff
@@ -276,7 +323,7 @@ def generate_translation(request):
     # Check if the cached translation is valid and return if it is valid
     translation = (
         Translation.objects.filter(
-            transcript=transcript_id, target_lang=target_lang, user=request.user.id
+            transcript=transcript_id, target_language=target_language, user=request.user.id
         )
```

Comment:
This endpoint is basically used to retrieve `MACHINE_GENERATED` translation (if already present; if not, generate).
So we can filter based on that type also.

Also, we should remove the filter based on user.

Reply:
The above suggestions are in line with what we are already doing in the `create_transcription` endpoint.

Reply:
Yeah, that is what we do in `transcript`. I will make it consistent across both the apps. Another point is to rename the transcript function from `create_transcript` to `generate_transcript` to ensure consistency.

Reply:
Sure, please do, thanks!